packages/components/src/components/autocomplete/autocomplete.jsx

```diff
@@ -44,7 +45,9 @@ const Autocomplete = React.memo(props => {
         onHideDropdownList,
         onItemSelection,
         onScrollStop,
+        onShowDropdownList,
         should_filter_by_char,
+        show_list = false,
```

Comment:
What does `should_show_list` do and why are these not interchangeable?

Reply:
since autocomplete is handling showing and hiding the dropdown list by using the function hideDropdownList which also takes props passed from onHideDropdownList, i had to change how we show and hide the list because of how they wanted the input to behave. When the user clicks the close icon, they want to show the dropdown immediately:

<img width="739" alt="Screenshot 2022-11-30 at 3 59 09 PM" src="https://user-images.githubusercontent.com/103412909/204739794-d20b4419-2de7-413c-836b-417708144226.png">
<img width="712" alt="Screenshot 2022-11-30 at 3 59 04 PM" src="https://user-images.githubusercontent.com/103412909/204739802-ba60b359-dc01-4d60-988c-ec0a8f13dcd3.png">

but the dropdown list only appears once the user clicks the input, hence why i needed to use show_list to control this aspect. But then there were also issues when you select one payment method, a new autocomplete would display below but immediately shows the dropdown list even though we haven't clicked on it, and that isn't the expected behaviour. So I had to also control how and when to hide the list... and then there was the case above as well when clicking outside of the component, since i forced it to show when the user didn't focus/clicked on autocomplete, after clicking the close icon, the component already has a handler for that, but when it has already been focused on, so the outside click wouldn't register.